### DISCUSSION TITLE
drag and drop reordering ReorderNestedModuleItems so slow on larger datasets its unuasable

### DISCUSSION THREAD

**Discussion by futurewebsites** [area17/twill#2138] [General]
Hi guys,

I have added nested sets to a category model which contains a category tree up to 5 levels deep the first issue I had is that in the standard queue nothing saved as it always timed out so I added an override in the repository to force the job to a longer running queue 

````
public function __construct(Category $model)
	{
		$this->model = $model;

		$this->reorderNestedModuleItemsJobQueue = 'long-running';
	}
````
	
this is not ideal and the jobs are still timing out at 23+ seconds but a far bigger issue is the front end UI is so slow and laggy its actually near on impossible to drag the categories into the correct order. 

If anyone has any suggestions to improve performance for larger datasets it would be greatly appreciated.

**Comment by ifox** [comment]
Hi @futurewebsites thanks for reporting this. 

How many categories are we talking about that you'd qualify as "large"?

The goal of the queued job is actually to make sure the UI can stay interactive. So it seems like you are reporting 2 problems here, one being a JS performance issue, the other one being the job being too slow.

**Reply by futurewebsites** [reply]
Hi @iFox yes I think they are 2 separate issues, currently there are 2,515 rows in the categories table with various nesting levels which isn't massive but enough to make it unusable.
